File: packages/oasys_core/lib/models/requests/create_trail_request.dart
class CreateTrailRequest {
  CreateTrailRequest({
    required this.name,
    required this.description,
    required this.city,
    required this.durationMinutes,
    required this.distanceKm,
    required this.difficulty,
  });

  final String name;
  final String description;
  final String city;
  final int durationMinutes;
  final double distanceKm;
  final int difficulty;

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'name': name,
      'description': description,
      'city': city,
      'durationMinutes': durationMinutes,
      'distanceKm': distanceKm,
      'difficulty': difficulty,
    };
  }
}

File: backend/Oasys.Application/DTOs/Request/CreateTrailRequest.cs
using System.ComponentModel.DataAnnotations;
using Oasys.Core.Enums;

namespace Oasys.Application.DTOs.Request;

public sealed class CreateTrailRequest
{
    [Required(ErrorMessage = "Naziv je obavezan.")]
    [StringLength(120, MinimumLength = 2, ErrorMessage = "Naziv mora imati izmedu 2 i 120 karaktera.")]
    public string Name { get; set; } = string.Empty;

    [Required(ErrorMessage = "Opis je obavezan.")]
    [StringLength(2000, MinimumLength = 10, ErrorMessage = "Opis mora imati izmedu 10 i 2000 karaktera.")]
    public string Description { get; set; } = string.Empty;

    [Required(ErrorMessage = "Grad je obavezan.")]
    [StringLength(120, MinimumLength = 2, ErrorMessage = "Grad mora imati izmedu 2 i 120 karaktera.")]
    public string City { get; set; } = string.Empty;

    [Range(1, 10080, ErrorMessage = "Trajanje mora biti izmedu 1 i 10080 minuta.")]
    public int DurationMinutes { get; set; }

    [Range(typeof(decimal), "0.1", "1000", ErrorMessage = "Duzina mora biti izmedu 0.1 i 1000 kilometara.")]
    public decimal DistanceKm { get; set; }

    [Required(ErrorMessage = "Tezina staze je obavezna.")]
    public TrailDifficulty Difficulty { get; set; }
}
